src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/RetrievePaymentAccountTest.java

```diff
@@ -0,0 +1,178 @@
+        userRoles.add(roleUnderTest);
+
+        String userEmail = randomAlphabetic(5).toLowerCase() + "@hotmail.com";
+        String lastName = "someLastName";
+        String firstName = "someName";
+
+        bearerTokenForUser = professionalApiClient.getMultipleAuthHeadersExternal(roleUnderTest, firstName, lastName, userEmail);
+
+        NewUserCreationRequest userCreationRequest = aNewUserCreationRequest()
+                .firstName(firstName)
+                .lastName(lastName)
+                .email(userEmail)
+                .roles(userRoles)
+                .jurisdictions(OrganisationFixtures.createJurisdictions())
+                .build();
+        Map<String, Object> newUserResponse = professionalApiClient.addNewUserToAnOrganisation(orgIdentifier, hmctsAdmin, userCreationRequest);
+
+        userId = (String) newUserResponse.get("userIdentifier");
+
+        return bearerTokenForUser;
+    }
+
+    @Test
+    public void rdcc117_ac1_pui_finance_manager_can_retrieve_a_list_of_pbas_of_a_given_organisation() {
+        Map<String, Object> response = professionalApiClient.retrievePbaAccountsForAnOrganisationExternal(HttpStatus.OK, generateBearerTokenForUser(puiFinanceManager));
+        assertThat(response.get("paymentAccount")).asList().hasSize(1);
+    }
+
+    @Test
+    public void rdcc117_ac2_pui_organisation_manager_can_retrieve_a_list_of_pbas_of_a_given_organisation() {
+        Map<String, Object> response = professionalApiClient.retrievePbaAccountsForAnOrganisationExternal(HttpStatus.OK, generateBearerTokenForUser(puiOrgManager));
+        assertThat(response.get("paymentAccount")).asList().hasSize(1);
+    }
+
+    @Test
+    public void rdcc117_ac3_user_without_appropriate_permission_cannot_retrieve_a_list_of_pbas_of_a_given_organisation() {
+        Map<String, Object> response = professionalApiClient.retrievePbaAccountsForAnOrganisationExternal(HttpStatus.FORBIDDEN, generateBearerTokenForUser(puiCaseManager));
+        assertThat(response.get("errorMessage")).isNotNull();
+        assertThat(response.get("errorMessage")).isEqualTo("9 : Access Denied");
+    }
+
+    @Test
+    public void rdcc117_ac4_pui_organisation_or_finance_manager_without_active_status_cannot_retrieve_a_list_of_pbas() {
+
+        String email = randomAlphabetic(10) + "@somewhere.com".toLowerCase();
+        Map<String, Object> createOrgResponse = professionalApiClient.createOrganisation();
+        OrganisationCreationRequest organisationCreationRequest = createOrganisationRequest().superUser(aUserCreationRequest()
+                .firstName("some-fname")
+                .lastName("some-lname")
+                .email(email)
+                .jurisdictions(createJurisdictions())
+                .build()).build();
+        orgIdentifier = (String) createOrgResponse.get("organisationIdentifier");
+        professionalApiClient.updateOrganisation(orgIdentifier, hmctsAdmin);

```

Comment:
Seeing as this is repeated in most/all tests, you could, to save space and make the tests easier to read, move this selected block into a private method in this test class - and call this method to generate the new user 😄